src/HDF5.jl

```diff
@@ -1309,8 +1359,8 @@ end
 # Clean up string buffer according to padding mode
 function unpad(s::String, pad::Integer)
     if pad == H5T_STR_NULLTERM
-        v = search(s, '\0')
-        v == 0 ? s : s[1:v-1]
+        v = Compat.findfirst(equalto('\0'), s)
```

Comment:
Does this not need  Compat v0.55.2 as a minimum?

Reply:
Maybe just add the import `Compat: StingVector, findfirst` to the top of the file like `StringVector`? (not required, but I just thought  it might be cleaner, when these imports are dropped for 0.6)

Reply:
I've bumped the Compat version to v0.56 (this will probably be required either way, once 0.7 arrives) and explicitly listed all uses of Compat.

Reply:
I think I quoted the wrong minimum. It should be  v0.53.0https://github.com/JuliaLang/Compat.jl/commit/8cca897ff1bf251bc2fb6e806ec481cfc478de26    I updated the PR.  With minimums we need to be sure to put the minimums, so to not unnecessarily force and update on users (best practice).

Thanks again